actionpack/test/abstract/callbacks_test.rb

```diff
@@ -158,6 +158,56 @@ def setup
       end
     end
 
+    class CallbacksWithReusedConditions < ControllerWithCallbacks
+      def index
+        self.response_body = @list.join(", ")
+      end
+
+      def public_data
+        @authenticated = "false"
+        self.response_body = @authenticated
+      end
+
+      private
+        def list
+          @list = ["Hello", "World"]
+        end
+
+        def authenticate
+          @list ||= []
+          @authenticated = "true"
+        end
+    end
+
+    class TestCallbacksWithReusedConditions < ActiveSupport::TestCase
+      def setup
+        @controller = CallbacksWithReusedConditions.new
+      end
+
+      def define_callbacks
+        options = { only: :index }
+        CallbacksWithReusedConditions.before_action :list, options
+        CallbacksWithReusedConditions.before_action :authenticate, options
+      end
+
+      test "callbacks can be defined with reused conditions" do
+        define_callbacks
```

Comment:
This is an odd-looking test: `assert_nothing_raised { define_callbacks }` would be slightly better, but I also think it's wrong that we're defining the callbacks on the same class in multiple tests. We could create an anonymous subclass in the setup method, but how about adding the callbacks when the controller is defined, and removing the "it doesn't crash" test entirely?

Reply:
Sure. I can do that.

Reply:
Done.